js/capmap.js

```diff
@@ -982,316 +951,99 @@ function doCAP(dom) {
-        iconUrl: symbolPath + 'temperature.png',
-        iconSize: [alertOptions.iconWidth, alertOptions.iconHeight],
-        iconAnchor: [alertOptions.iconWidth / 2 + xDisplacement, alertOptions.iconWidth / 2],
-        popupAnchor: [0, 0]
-      })
-    } else if (~eventRaw.indexOf('snow')) {
-      var icon = L.icon({
-        iconUrl: symbolPath + 'snow.png',
-        iconSize: [alertOptions.iconWidth, alertOptions.iconHeight],
-        iconAnchor: [alertOptions.iconWidth / 2 + xDisplacement, alertOptions.iconWidth / 2],
-        popupAnchor: [0, 0]
-      })
-      // Rainfall Icon
-    } else if (~eventRaw.indexOf('rain') || ~eventRaw.indexOf('shower')) {
-      var icon = L.icon({
-        iconUrl: symbolPath + 'rainfall.png',
-        iconSize: [alertOptions.iconWidth, alertOptions.iconHeight],
-        iconAnchor: [alertOptions.iconWidth / 2 + xDisplacement, alertOptions.iconWidth / 2],
-        popupAnchor: [0, 0]
-      })
-    } else if (~eventRaw.indexOf('icing')) {
-      var icon = L.icon({
-        iconUrl: symbolPath + 'snow.png',
-        iconSize: [alertOptions.iconWidth, alertOptions.iconHeight],
-        iconAnchor: [alertOptions.iconWidth / 2 + xDisplacement, alertOptions.iconWidth / 2],
-        popupAnchor: [0, 0]
-      })
-    }
-
-    // Placeholder for sleet
-    else if (~eventRaw.indexOf('sleet')) {
-      var icon = L.icon({
-        iconUrl: symbolPath + 'sleet.png',
-        iconSize: [alertOptions.iconWidth, alertOptions.iconHeight],
-        iconAnchor: [alertOptions.iconWidth / 2 + xDisplacement, alertOptions.iconWidth / 2],
-        popupAnchor: [0, 0]
-      })
-    }
+    function createIcon(symbolPath, alertOptions, eventRaw, windSpeed, windDirection, waveHeight, swellHeight, surfHeight, eventSelector) {
+      // Set default icon
+      let iconUrl = symbolPath + 'gale.png';  // Fallback icon
+      let iconSize = [alertOptions.iconWidth, alertOptions.iconHeight];
+      let iconAnchor = [alertOptions.iconWidth / 2 + alertOptions.xDisplacement || 0, alertOptions.iconWidth / 2];
+      let popupAnchor = [0, 0];
+
+      // Determine icon URL based on conditions
+      if (windSpeed > 0) {
+        iconUrl = alertOptions.numberIcons
+          ? `${symbolPath}wind.php?speed=${windSpeed}&direction=${windDirection}`
+          : symbolPath + 'wind-speed.png';
+      } else if (waveHeight > 0) {
+        iconUrl = alertOptions.numberIcons
+          ? `${symbolPath}wave.php?height=${waveHeight}`
+          : symbolPath + 'wave-height.png';
+      } else if (swellHeight > 0) {
+        iconUrl = alertOptions.numberIcons
+          ? `${symbolPath}wave.php?height=${swellHeight}`
+          : symbolPath + 'swell-height.png';
+      } else if (surfHeight > 0) {
+        iconUrl = alertOptions.numberIcons
+          ? `${symbolPath}wave.php?height=${surfHeight}`
+          : symbolPath + 'surf-height.png';
+      } else {
+        // Handle events based on eventRaw or eventSelector
+        let eventMapping = [
```

Comment:
Maybe consider sorting alphabetically by match, or perhaps by icon. Or maybe if there is some more generic event type to be used here. E.g. sort wave and tsunami and high tide and other "watery evens" close to each other.

Reply:
Some sort of manual arrangement would be practical and i'll think about a good solution. Alphabetical order doesn't work because, for example, in cases of flooding, the "flood" warning would be the first, and the function pick it instead of checking for exact warning types after that.